packages/flutter_tools/lib/src/cache.dart

```diff
@@ -92,8 +94,8 @@ class Cache {
   /// [rootOverride] is configurable for testing.
   /// [artifacts] is configurable for testing.
   Cache({
-    Directory rootOverride,
-    List<ArtifactSet> artifacts,
+    @visibleForOverriding Directory rootOverride,
+    @visibleForOverriding List<ArtifactSet> artifacts,
```

Comment:
Note sure if this is the right annotation?

Reply:
@visibleForTesting if it should only be used in test files. `@visibleForOverriding` I'm not really sure what that means that is different from `@protected`

Reply:
Yeah I think I want `@protected`.